src/Containers/packaging/build/Microsoft.NET.Build.Containers.targets

```diff
@@ -216,11 +217,69 @@
+      <_RequiredContainerPublishTargets>Publish;PublishContainer</_RequiredContainerPublishTargets>
+    </PropertyGroup>
+
+    <!-- Strategy here is that we will figure out what project(s) to build the containerization targets(s) for
+         based on project state. We use `AdditionalProperties` to customize the outputs of each of the builds. -->
+
+    <!-- Properties set here:
+      * TargetFramework - multitargeting - changes inference for base image based on TFM
+      * ContainerImageTag - this is a rough cut
+      * ContainerRuntimeIdentifier - if we're building for a specific RID, we need to set this so that the
+                                     containerization targets know what RID to build for
+      * RuntimeIdentifier - if we're building for a specific RID, we need to set this so that we get optimized
+                            RID-specific assets in the publish output
+
+      NOTE: we could get away with setting `RuntimeIdentfier` here to control `ContainerRuntimeIdentifier` inference
+            but this is also nice and explicit.
+     -->
+
+    <!-- TFMs but no TF -> multitarget, making image for each TFM -->
+    <ItemGroup Condition="'$(TargetFrameworks)' != ''
+                          and '$(TargetFramework)' == ''" >
+      <_TFMItems Include="$(TargetFrameworks)" />
+      <_SingleContainerPublish Include="$(MSBuildProjectFullPath)"
+        AdditionalProperties="TargetFramework=%(_TFMItems.Identity);
+                              ContainerImageTag=$([MSBuild]::GetTargetFrameworkVersion('%(_TFMItems.Identity)', 2))" />
+    </ItemGroup>
+
+    <!-- TF but no TFMs -> single image (aka the default pathway) up until now -->
+    <ItemGroup Condition="'$(TargetFramework)' != ''
+                          and '$(RuntimeIdentifiers)' == ''">
+        <_SingleContainerPublish Include="$(MSBuildProjectFullPath)" />
+    </ItemGroup>
+
+    <!-- TF with RIDs -> multi-arch, single image per arch -->
+    <ItemGroup Condition="'$(TargetFramework)' != ''
+                          and '$(RuntimeIdentifiers)' != ''">
+      <_RIDItems Include="$(RuntimeIdentifiers)" />
+    </ItemGroup>
+    <MapRidToDockerPlatform Condition="'$(TargetFramework)' != ''
+                                      and '$(RuntimeIdentifiers)' != ''"
+                            RuntimeIdentifiers="@(_RIDItems)">
+      <Output TaskParameter="ModifiedRuntimeIdentifiers" ItemName="_RIDItemsWithDockerPlatformTag" />
+    </MapRidToDockerPlatform>
+    <ItemGroup Condition="'$(TargetFramework)' != ''
+                          and '$(RuntimeIdentifiers)' != ''">
+      <_SingleContainerPublish Include="$(MSBuildProjectFullPath)"
+          AdditionalProperties="ContainerRuntimeIdentifier=%(_RIDItemsWithDockerPlatformTag.Identity);
+                                RuntimeIdentifier=%(_RIDItemsWithDockerPlatformTag.Identity);
+                                ContainerImageTag=$(Version)-%(_RIDItemsWithDockerPlatformTag.DockerPlatformTag);" />
```

Comment:
Here's an interesting consequence of moving to `latest` for image tag inference as a default. Before, `$(Version)` was used as the default tag, which meant we had something useful to hang a platform-specific bit off of. For example, `1.0.0-amd64` would make sense as a container image tag.

Now that we default to latest, you would see things like `latest-amd64`, which I have never seen as a pattern or practice before.  What I would _like_ to do is just set something like `ContainerImageTagSuffix` here and flow `Version` along, but I think there's even a bug with that - `$(Version)` at the point that this target is run may not be calculated yet. For example my OSS projects generate Version in a target that's hooked up to `PrepareForBuild` as a dependency. To me, this means we need to find a way to inject this on a per-image basis. I'm not sure how to tackle that one. Love to have any ideas.

Reply:
@rainersigwald hate to ping, but I had a design question that I'd love your input on if you have a moment.